src/chart/generateCategoricalChart.tsx

```diff
@@ -1569,11 +1577,12 @@ export const generateCategoricalChart = ({
 
     /**
      * The handler of mouse entering a graphical item, such as bar, pie, scatter, funnel, ...
+     * This handler is used for `tooltipEventType: item` and both `Tooltip.trigger: hover` and `Tooltip.trigger: click`
      * @param el The active graphical element
      * @param index 0-based index of the active graphical element
-     * @returns undefined
+     * @return undefined
      */
-    handleItemMouseEnter = (el: any, index: number) => {
+    handleItemMouseEnter: ActivateTooltipAction = (el, index) => {
```

Comment:
at this point what is the value of even having these here? Would all hell break loose if we just said lets define them in each graphical item (kind of like what we're doing above anyways) and be done with it? we could set tooltip state from there no?

Reply:
generateCategoricalChart still needs the tooltip data in its internal state. It's using it for synchronising different charts and possibly other things too.

Reply:
ah right, forgot about synchronization